src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -98,37 +111,54 @@ synchronized void createTable(String tableName, List<Field<?>> columns, String r
                                 String referenceTablePrimaryKeyFieldName) throws SQLException {
     Field foreignKeyField = DSL.field(referenceTablePrimaryKeyFieldName, Integer.class);
     columns.add(foreignKeyField);
-    Table referenceTable = DSL.table(referenceTableName);
-    CreateTableConstraintStep constraintStep = create.createTable(tableName)
-        .column(DSL.field(getPrimaryKeyColumnName(tableName) + PRIMARY_KEY_AUTOINCREMENT_POSTFIX))
-        .columns(columns)
-        .constraints(DSL.constraint(foreignKeyField.getName() + "_FK").foreignKey(foreignKeyField)
-            .references(referenceTable, DSL.field(referenceTablePrimaryKeyFieldName)));
 
-    LOG.debug("table with fk created: {}", constraintStep.toString());
     try {
+      LOG.debug("Trying to create a summary table: {} that references {}", tableName, referenceTableName);
+      Table referenceTable = DSL.table(referenceTableName);
+      CreateTableConstraintStep constraintStep = create.createTable(tableName)
```

Comment:
I assume creating the SQL statement itself does not throw any exception. Can we leave this part out of the try block ?

Reply:
The DSLContext can be uninitialized if we are not using synchronization between the threads. I saw a case where the table was added to hastable but the columns for the table were missing. We gather some date to create the createStatement and it the gathering part we can run into exceptions.
These cases are fixed now but it would be good to catch and log exception. But you are right, the scope of exceptions is larger and so I have added code to catch exception and not just DataAcessException.